File: Python/Arrays/L_maxElemAftDecr&Rearr.py
from typing import List


def maximumElementAfterDecrementingAndRearranging(arr: List[int]) -> int:
        arr.sort()
        max_val = 1
        for i in range(1, len(arr)):
            if arr[i] > max_val:
                max_val += 1
        return max_val

File: cpp/Array/L_maxElemAftDecr&Rearr.cpp
#include <bits/stdc++.h>
using namespace std;

int maximumElementAfterDecrementingAndRearranging(vector<int> &arr)
{
    sort(arr.begin(), arr.end());
    int maxVal = 1;

    for (int i = 1; i < arr.size(); i++)
    {
        if (arr[i] > maxVal)
        {
            maxVal += 1;
        }
    }

    return maxVal;
}